src/lib/controllers/ui-controller.js

```diff
@@ -870,25 +970,29 @@ export default class UIController {
   }
 
   updateWordUsageExamples (wordUsageExamplesData) {
-    this.addMessage(this.api.l10n.getMsg('TEXT_NOTICE_WORDUSAGE_READY'))
+    this.store.commit('ui/addMessage', this.api.l10n.getMsg('TEXT_NOTICE_WORDUSAGE_READY'))
     this.store.commit('app/setWordUsageExamplesData', wordUsageExamplesData)
   }
 
   clear () {
-    this.store.commit(`app/resetInflData`)
+    this.store.commit(`app/resetStatusData`)
+    this.store.commit(`app/resetDefDataReady`)
+    this.resetInflData()
     this.store.commit(`app/resetTreebankData`)
-    if (this.hasUiModule('panel')) { this.getUiModule('panel').vi.clearContent() }
-    if (this.hasUiModule('popup')) { this.getUiModule('popup').vi.clearContent() }
+    this.store.commit(`ui/resetNotification`)
+    this.store.commit(`ui/resetMessages`)
     return this
   }
 
-  // TODO: Is this ever called?
   open () {
-    if (this.contentOptions.items.uiType.currentValue === this.options.uiTypePanel) {
-      if (this.api.ui.hasModule('panel')) { this.api.ui.openPanel() }
+    if (this.api.ui.hasModule('panel') && this.api.ui.getModule('panel').config.panelComponent === 'compactPanel') {
```

Comment:
this test doesn't work. Maybe config.panelComponent is something you have since refactored? In any event, what do you think about changing this to just test if api.ui.getModule('popup') is false? Since right now the panel and popup are the only two ways to show results, if we don't have a popup we're going to always want to show the panel I think.

Reply:
A good catch! I've updated it to reflect the current architecture